packages/vscode-tailwindcss/src/extension.ts

```diff
@@ -301,363 +289,216 @@ export async function activate(context: ExtensionContext) {
 
     let serverOptions: ServerOptions = {
       run: {
         module,
         transport: TransportKind.ipc,
-        options: { execArgv: inspectPort === null ? [] : [`--inspect=${inspectPort}`] },
+        options: {
+          execArgv: inspectPort === null ? [] : [`--inspect=${inspectPort}`],
+        },
       },
       debug: {
         module,
         transport: TransportKind.ipc,
         options: {
-          execArgv: ['--nolazy', `--inspect=${6011 + clients.size}`],
+          execArgv: ['--nolazy', `--inspect=6011`],
         },
       },
     }
 
     let clientOptions: LanguageClientOptions = {
-      documentSelector: languages.get(folder.uri.toString()).map((language) => ({
-        scheme: 'file',
-        language,
-        pattern: normalizePath(`${folder.uri.fsPath.replace(/[\[\]\{\}]/g, '?')}/**/*`),
-      })),
+      documentSelector: documentFilters,
       diagnosticCollectionName: CLIENT_ID,
-      workspaceFolder: folder,
       outputChannel: outputChannel,
       revealOutputChannelOn: RevealOutputChannelOn.Never,
       middleware: {
-        provideCompletionItem(document, position, context, token, next) {
-          let workspaceFolder = Workspace.getWorkspaceFolder(document.uri)
-          if (workspaceFolder !== folder) {
-            return null
-          }
-          return next(document, position, context, token)
-        },
-        provideHover(document, position, token, next) {
-          let workspaceFolder = Workspace.getWorkspaceFolder(document.uri)
-          if (workspaceFolder !== folder) {
-            return null
-          }
-          return next(document, position, token)
-        },
-        handleDiagnostics(uri, diagnostics, next) {
-          let workspaceFolder = Workspace.getWorkspaceFolder(uri)
-          if (workspaceFolder !== folder) {
-            return
-          }
-          next(uri, diagnostics)
-        },
-        provideCodeActions(document, range, context, token, next) {
-          let workspaceFolder = Workspace.getWorkspaceFolder(document.uri)
-          if (workspaceFolder !== folder) {
-            return null
-          }
-          return next(document, range, context, token)
-        },
         async resolveCompletionItem(item, token, next) {
+          let editor = Window.activeTextEditor
+          if (!editor) return null
+
           let result = await next(item, token)
-          let selections = Window.activeTextEditor.selections
+          if (!result) return result
+
+          let selections = editor.selections
+          let edits = result.additionalTextEdits || []
+
           if (
-            result['data'] === 'variant' &&
-            selections.length > 1 &&
-            result.additionalTextEdits?.length > 0
+            selections.length <= 1 ||
+            edits.length === 0 ||
+            // @ts-expect-error: use of private API / implementation details of our language server
```

Comment:
This gives me an error in my editor and while creating a build that it is unused. I guess this is not an actual error anymore 🤔

Reply:
(Deleting this, will result in Prettier changes)

Reply:
Oh actually, if you change `result['data']` to `result.data` it _does_ show up as an error and the `// @ts-expect-error` is required.

Reply:
VScode shows this as an error but for some reason Zed does not 🤷‍♂️